juicebox_scripts/juicebox_assembly_purger.py

```diff
@@ -0,0 +1,131 @@
+#!usr/bin/env python
+'''
+Brad Nelson
+May 29, 2019
+Phase Genomics
+
+juicebox_scripts/juicebox_assembly_purger.py
+
+This script takes an assembly file and outputs an assembly file
+without the specified contigs.
+
+Copyright 2019, Phase Genomics Inc. All rights reserved.
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU Affero General Public License as
+published by the Free Software Foundation, either version 3 of the
+License, or (at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Affero General Public License for more details.
+
+You should have received a copy of the GNU Affero General Public License
+along with this program.  If not, see https://www.gnu.org/licenses/agpl-3.0.en.html
+'''
+
+from __future__ import print_function
+import argparse
+
+def parse_args():
+    parser = argparse.ArgumentParser()
+    parser.add_argument("input_assembly")
+    parser.add_argument("output_assembly")
+    parser.add_argument("--exclude_contigs",
+                        default=None,
+                        nargs="+",
+                        help="Names of contigs to exclude")
+    parser.add_argument("--exclude_file",
+                        default=None,
+                        help="Path to file of contigs to exclude (with contig names in "
+                             "first whitespace-delimited column, one per line)")
+    args = parser.parse_args()
+
+    if args.exclude_contigs is None and args.exclude_file is None:
+        raise ValueError("Error: exclude_contigs or exclude_file must be specified")
+
+    return vars(args)
+
+def get_exclude(contigs, file):
+    """Takes a list and/or file of contig names and returns a set of contig names to exclude.
+    Assumes contigs and file are not both None.
+    """
+    exclude = set([])
+    if contigs is not None:
+        for name in contigs:
+            exclude.add(name)
+    if file is not None:
+        with open(file, 'r') as input:
+            for line in input:
+                name = line.split()[0]
+                exclude.add(name)
+
+    return exclude
+
+def filter_assembly(exclude, input_assembly, output_assembly, **kwargs):
+    """Take an input assembly file and create an output assembly file
+    without the contigs specified in the exclude set.
+    """
+    with open(input_assembly, "r") as input, \
```

Comment:
`input` is reserved, as is showing up in syntax highlighting. might make sense to change var name.

Reply:
Yeah this would be good to change

Reply:
Good call. Fixed.